From 06d1894f3eb6325a26cf3d0264896bc211e6b2e2 Mon Sep 17 00:00:00 2001
From: Leo Yang <Leo-Yang@quantatw.com>
Date: Wed, 12 Nov 2025 15:21:13 +0800
Subject: [PATCH] PSUSensor: add MP5998 support

Add device MP5998 reading support for YV5 project.

Tested:
Patch driver and build.
```
root@bmc:~# cat /sys/bus/i2c/devices/8-0046/hwmon/hwmon43/
curr1_crit        curr2_alarm       in1_input         in2_alarm         power1_alarm      power2_rated_max  uevent
curr1_crit_alarm  curr2_input       in1_label         in2_input         power1_input      subsystem/
curr1_input       curr2_label       in1_max           in2_label         power1_label      temp1_crit
curr1_label       curr2_rated_max   in1_max_alarm     in2_rated_max     power1_max        temp1_crit_alarm
curr1_max         device/           in1_min           name              power1_rated_max  temp1_input
curr1_max_alarm   in1_crit          in1_min_alarm     of_node/          power2_input      temp1_max
curr1_rated_max   in1_crit_alarm    in1_rated_max     power/            power2_label      temp1_max_alarm
```
---
 src/psu/PSUSensorMain.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/psu/PSUSensorMain.cpp b/src/psu/PSUSensorMain.cpp
index d91ad76..f3cca00 100644
--- a/src/psu/PSUSensorMain.cpp
+++ b/src/psu/PSUSensorMain.cpp
@@ -124,6 +124,7 @@ static const I2CDeviceTypeMap sensorTypes{
     {"MP2975", I2CDeviceType{"mp2975", true}},
     {"MP5023", I2CDeviceType{"mp5023", true}},
     {"MP5990", I2CDeviceType{"mp5990", true}},
+    {"MP5998", I2CDeviceType{"mp5998", true}},
     {"MP29612", I2CDeviceType{"mp29612", true}},
     {"MPQ8785", I2CDeviceType{"mpq8785", true}},
     {"NCP4200", I2CDeviceType{"ncp4200", true}},
-- 
2.47.3

